src/hooks/useUserLocation.js

```diff
@@ -0,0 +1,99 @@
+import { useEffect, useState } from 'react';
+import { Platform, Linking, PermissionsAndroid, Alert } from 'react-native';
+import Geolocation from 'react-native-geolocation-service';
+import { LOCATION_PERMISSIONS } from 'constants/permissions';
+import { IS_ANDROID, IS_IOS } from 'constants/platforms';
+import strings from 'localization';
+
+const useUserLocation = () => {
+  const [userLocation, setUserLocation] = useState({});
+
+  const hasPermissionIOS = async () => {
+    const openSetting = () => {
+      Linking.openSettings();
+    };
+    const status = await Geolocation.requestAuthorization('whenInUse');
+
+    if (status === LOCATION_PERMISSIONS.GRANTED) {
+      return true;
+    }
+
+    if (status === LOCATION_PERMISSIONS.DENIED || status === LOCATION_PERMISSIONS.DISABLED) {
+      Alert.alert(strings.LOCATION.alertText, '', [
+        { text: strings.LOCATION.goToSettings, onPress: openSetting },
+        { text: strings.LOCATION.closeButton },
+      ]);
+    }
```

Comment:
This is great! Maybe this `Alert` should be moved outside `hasPermissionIOS`. By reading its name, I think `hasPermissionIOS` should only return a bool without any side-effects.